net/tcp/tcp_accept.c

```diff
@@ -232,7 +232,7 @@ int psock_tcp_accept(FAR struct socket *psock, FAR struct sockaddr *addr,
     {
       /* Yes... get the address of the connected client */
 
-      ninfo("Pending conn=%p\n", state.acpt_newconn);
+      ninfo("Pending conn=%p\n", (void *)state.acpt_newconn);
```

Comment:
Any type pointer can implicitly convert to void *, is it too redundant to add cast here?

Reply:
If I compile the file by passing "-std=c89 -Wpedantic", the gcc produces "warning: format ‘%p’ expects argument of type ‘void *’, but argument 3 has type ‘struct tcp_conn_s *’ [-Wformat=]".

Reply:
%p for the non void pointer is used in many many place, and the usage is harmless too. The compiler warning is stupid sometime:(.

Reply:
I have updated the patch. ("warning: format ‘%p’ expects argument of type ‘void *’, but argument 3 has type ‘struct tcp_conn_s *’ [-Wformat=]" is not required to fix).